### PR TITLE
Allow to skip "empty serialization result" when serializing

### PR DESCRIPTION
**Pull request by goetas**
Fixes BC break introduced with #744 (and released with 1.6.1) by allowing to configure the serializer behavior when dealing with "empty" serialization result

[here](https://travis-ci.org/goetas-webservices/xsd2php/builds/221891989) some test suite failing with 1.6.1 & 1.6.2 while working with <=1.6.0 & this branch

"empty" means in this context: excluded by some exclusion-strategy, by nulls or empty objects with no properties

### CODE REVIEW

**Comment by Webonaute** [comment]
in JsonSerializationVisitor.php
line 162. It check for \ArrayObject. but if $v is a normal array, it doesnt skip it.

**Comment by goetas** [comment]
@Webonaute thanks! Fixed with https://github.com/schmittjoh/serializer/commit/34f5b67714c3925fba96e3d18ae2ed134f58cf57

**Comment by Tobion** [comment]
@goetas doesn't this fix #124?

**Comment by goetas** [comment]
good catch! thanks!

File: src/JMS/Serializer/JsonSerializationVisitor.php [hunk 1]
```diff
@@ -158,7 +158,9 @@ public function visitProperty(PropertyMetadata $metadata, $data, Context $contex
         $v = $this->accessor->getValue($data, $metadata);
 
         $v = $this->navigator->accept($v, $metadata->type, $context);
-        if (null === $v && $context->shouldSerializeNull() !== true) {
+        if ((null === $v && $context->shouldSerializeNull() !== true)
+            || ($v instanceof \ArrayObject && 0 === count($v) && $metadata->skipWhenEmpty)
```
Comment (Tobion) [inline src/JMS/Serializer/JsonSerializationVisitor.php]: So a plain empty array is not skipped as it checks for \ArrayObject? That's what I hoped for with #124

**Comment by goetas** [comment]
It works for arraya too

See https://github.com/schmittjoh/serializer/commit/34f5b67714c3925fba96e3d18ae2ed134f58cf57

**Pull request merged**